001-core/password.php

```diff
@@ -0,0 +1,63 @@
+<?php
+
+add_action( 'user_profile_update_errors', 'validate_current_password', 1, 3 );
+
+function validate_current_password( $errors, $update, $user ) {
+	check_admin_referer( 'update-user_' . $user->ID );
+
+	if ( ! isset( $_POST['pass1'] ) || empty( $_POST['pass1'] ) || ! $update ) {
+		return;
+	}
+
+	if ( ! isset( $_POST['current_pass'] ) || empty( $_POST['current_pass'] ) ) {
+		$errors->add( 'wrong_current_password', __( '<strong>ERROR</strong>: Please enter your current password.' ), array( 'form-field' => 'current_pass' ) );
+		return;
+	}
+
+	$error = wp_authenticate( $user->user_login, $_POST['current_pass'] );
+	if ( is_wp_error( $error ) ) {
+		$errors->add( 'wrong_current_password', __( '<strong>ERROR</strong>: The entered current password is not correct.' ), array( 'form-field' => 'current_pass' ) );
+	}
+}
+
+add_action( 'show_user_profile', 'add_current_password_field' );
+
+function add_current_password_field() { ?>
+	<table id="nojs-current-pass" class="form-table" role="presentation">
+		<tr>
+			<th><label for="current_pass"><?php _e( 'Current Password' ); ?></label></th>
+			<td>
+				<div id="current-password-confirm">
+					<input type="password" name="current_pass" id="current_pass" class="regular-text" value="" autocomplete="off" />
+					<p class="description">
+						<?php _e( 'Please, type your <strong>current password</strong> to change it for a new one' ); ?>.
+					</p>
+				</div>
+
+			</td>
+		</tr>
+	</table>
+	<?php 
+}
+
+add_action( 'admin_head', 'reposition_current_password_field' );
+
+function reposition_current_password_field() {
+	$screen = get_current_screen();
+	if ( 'profile' != $screen->id ) {
+		return;
+	} 
+	?>
+
+	<script type="text/javascript">
+		(function($) {
+			$(document).ready(function() {
```

Comment:
There's an effort to move away from jQuery in WordPress, so if this moving and styling of DOM elements can be done without jQuery, the better.

Also - what happens if someone doesn't have JS available?

Reply:
I'm pushing some changes to not use jQuery in this JS. If JS is not available, the repositioning will not happen and the form will look like this:

<img width="1147" alt="Screen Shot 2021-04-07 at 5 36 41 PM" src="https://user-images.githubusercontent.com/7188409/114162909-dd461a80-9929-11eb-8e63-97652499a338.png">